clang/runtime/dpct-rt/include/dpct/bindless_images.hpp

```diff
@@ -117,6 +137,341 @@ class image_mem_wrapper {
+        _res_is_buffer = true;
+        
+        D3D11_BUFFER_DESC desc;
+        ((ID3D11Buffer*)resource)->GetDesc(&desc);
+
+        res_width = desc.ByteWidth;
+
+        _res_size_bytes = desc.ByteWidth;
+        break;
+      }
+      case D3D11_RESOURCE_DIMENSION_TEXTURE1D: {
+        D3D11_TEXTURE1D_DESC desc;
+        ((ID3D11Texture1D*)resource)->GetDesc(&desc);
+
+        res_width = desc.Width;
+        res_arr_size = desc.ArraySize;
+        res_num_levels = (desc.MipLevels? desc.MipLevels: 1);
+
+        get_img_ch_info(
+            desc.Format, res_fomat_size_bytes, res_num_channels, img_ch_type);
+
+        _res_size_bytes = res_width * res_fomat_size_bytes * res_arr_size *
+                              res_num_levels;
+        break;
+      }
+      case D3D11_RESOURCE_DIMENSION_TEXTURE2D: {
+        D3D11_TEXTURE2D_DESC desc;
+        ((ID3D11Texture2D*)resource)->GetDesc(&desc);
+
+        res_width = desc.Width;
+        res_height = desc.Height;
+        res_arr_size = desc.ArraySize;
+        res_num_levels = (desc.MipLevels? desc.MipLevels: 1);
+
+        get_img_ch_info(
+            desc.Format, res_fomat_size_bytes, res_num_channels, img_ch_type);
+
+        _res_size_bytes = res_width * res_height * res_fomat_size_bytes *
+                              res_arr_size * res_num_levels;
+        break;
+      }
+      case D3D11_RESOURCE_DIMENSION_TEXTURE3D: {
+        D3D11_TEXTURE3D_DESC desc;
+        ((ID3D11Texture3D*)resource)->GetDesc(&desc);
+
+        res_width = desc.Width;
+        res_height = desc.Height;
+        res_depth = desc.Depth;
+        res_num_levels = (desc.MipLevels? desc.MipLevels: 1);
+
+        get_img_ch_info(
+            desc.Format, res_fomat_size_bytes, res_num_channels, img_ch_type);
+
+        _res_size_bytes = res_width * res_height * res_depth *
+                              res_fomat_size_bytes * res_num_levels;
+        break;
+      }
+      default:
+        throw std::runtime_error("Unsupported DX11 resource type!");
+        break;
+    }
+
+    assert(!(res_num_levels > 1 && res_arr_size > 1) &&
+        "Images with multiple mip and array levels are not supported!");
+    
+    sycl::ext::oneapi::experimental::image_type img_type = 
+        (res_num_levels > 1)?
+            sycl::ext::oneapi::experimental::image_type::mipmap:
+            sycl::ext::oneapi::experimental::image_type::standard;
+
+    _img_desc = sycl::ext::oneapi::experimental::image_descriptor{
+                    sycl::range<3>{res_width, res_height, res_depth},
+                    res_num_channels, img_ch_type, img_type, res_num_levels,
+                    res_arr_size};
+  }
+
+  // Helper function to create shared handle for DXD11 resource
+  HANDLE create_shared_handle(ID3D11Resource *resource) {
+    HANDLE win_nt_handle = nullptr;
+
+    IDXGIResource1* pDXGIResource1;
+    resource->QueryInterface(__uuidof(IDXGIResource1), reinterpret_cast<void**>(&pDXGIResource1));
```

Comment:
Does icpx/clang support the operator `__uuidof`?

Reply:
Yes, code ran as expected without errors on a Windows machine when tested with clang++ compiler